src/cli.js

```diff
@@ -78,7 +82,8 @@ const cli = argv => {
       const outCode = convert(inCode, options);
 
       if (program.write) {
-        const outFile = file.replace(/\.js$/, ".ts");
+        const extension = detectJsx(outCode) ? ".tsx" : ".ts";
```

Comment:
```suggestion
        const extension = detectJsx(inCode) ? ".tsx" : ".ts";
```

The detectJsx function is using babel settings for flow, and does not support TypeScript-specific syntax.

Reply:
Ah, that makes sense. 👍